### PR TITLE
Happy-Thoughts-API Ellen

### PR DESCRIPTION
**Pull request by ellenbjo**
💌back-end: https://share-happy-thought.herokuapp.com/thoughts 

💌 front-end: https://happyhappythoughts.netlify.app/ 

Things I'd like to have clarified or explained in more detail:
How to connect pagination from back-end to front-end🙈

### CODE REVIEW

File: models/thought.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
+export default Thought
```
Comment (ClaireCaudwell) [inline models/thought.js:20]: Nice to see that you created a seperate component for your model 👌

File: server.js [hunk 2]
```diff
@@ -2,25 +2,91 @@ import express from 'express'
...
 
-// Start defining your routes here
+app.use((req, res, next) => {
+  if(mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'service unavailable' })
+  }
+})
+
```
Comment (ClaireCaudwell) [inline server.js:27]: Is this the check that is done to check if the database is down?

File: server.js [hunk 3]
```diff
@@ -2,25 +2,91 @@ import express from 'express'
...
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
```
Comment (ClaireCaudwell) [inline server.js:30]: I like how you found and implemented this listEndpoints library to list all the endpoints in the API 👌

File: server.js [hunk 4]
```diff
@@ -2,25 +2,91 @@ import express from 'express'
...
+
+    res.status(200).json({
+      thoughts,
+      totalPages: Math.ceil(count / limit),
+      currentPage: page
+    })
+  } catch (error) {
+    res.status(400).json({success: false, message: error.message})
+  } */
+})
```
Comment (ClaireCaudwell) [inline server.js:56]: Good job for trying to implement pagination. Hope you can get some help to make this work 😊

File: server.js [hunk 5]
```diff
@@ -2,25 +2,91 @@ import express from 'express'
...
+//not used in front-end at the moment.
+app.delete('/thoughts/:id', async (req, res) => {
+  try {
+    await Thought.deleteOne({ _id: req.params.id })
+    res.status(200).json({ success: true })
+  } catch (error) {
+    console.log(error)
+    res.status(400).json({ success: false, error: error })
+  }
+})
```
Comment (ClaireCaudwell) [inline server.js:77]: Nice to see that you implemented a delete request endpoint, especially as I need to implement a delete endpoint in my project. And to see that mongoose has a method called deleteOne 👍

File: server.js [hunk 6]
```diff
@@ -2,25 +2,91 @@ import express from 'express'
...
+
+//updateOne- first object matches the object in the database, second what object should be changed
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  try {
+    const updatedThought = await Thought.updateOne(
+      { _id: req.params.thoughtId },
+      { $inc: { hearts: 1 } }
```
Comment (ClaireCaudwell) [inline server.js:84]: I don't know if this really makes any difference, but I wrote +1. I tried out the frontend and the same heart could be incremented more than once, so it works really well. Maybe the teacher that will get this code review can clarify if there's any difference between writing 1 or +1 for the heart count 😊

**Review by ClaireCaudwell — COMMENTED** [review]
Hi Ellen 😊

Your API endpoints looks great and work as expected in the frontend. Interesting to see how you've implemented a delete endpoint and to learn about the listEndpoints library you imported. 

I have a comment regarding the heart count that it would be nice to know if there is any different writing 1 or +1 from the code coach. Other than that looks great! Fingers crossed you can get the pagination to work at some point 💖

**Review by KarolCislowski — APPROVED** [review]
Great project 🥇

**Pull request closed without merging**